ui-core/src/components/inputs/FieldWrapper.tsx

```diff
@@ -11,15 +11,27 @@ export type FieldWrapperProps = {
   id: string;
   label?: string;
   hint?: string;
-  required?: boolean;
   disabled?: boolean;
   statusWithMessage?: StatusWithMessage;
   statusIconPosition?: 'next-to-field' | 'before-status-message';
   small?: boolean;
   children?: React.ReactNode;
   className?: string;
   onCloseStatusMessage?: () => void;
-};
+  /**
+   * "narrow" prop corresponds to an input without any wrapper padding.
+   * Should not be used with required or statusWithMessage with no tooltip.
+   */
+} & (
+  | {
+      required: boolean;
+      narrow: undefined;
+    }
+  | {
+      narrow: boolean;
+      required: undefined;
+    }
```

Comment:
This forbids things such as:

```ts
{ required: false, narrow: true }
{ narrow: false, required: false }
```

I think these should be allowed?

Proposal:

```ts
  | {
      required: true;
      narrow?: false;
    }
  | {
      narrow: true;
      required?: false;
    }
```